internal/controller/pipelinerollout/pipelinerollout_progressive.go

```diff
@@ -108,54 +119,54 @@ func (r *PipelineRolloutReconciler) AssessUpgradingChild(
 	if err != nil {
 		return assessment, reasonFailure, err
 	}
+
 	// if we fail even once, that's considered failure
 	if assessment == apiv1.AssessmentResultFailure {
 		// set AssessmentEndTime to now and return failure
+		numaLogger.Debugf("Assessment failed for upgrading child %s", existingUpgradingChildDef.GetName())
 		_ = progressive.UpdateUpgradingChildStatus(pipelineRollout, func(status *apiv1.UpgradingChildStatus) {
-			assessmentEndTime := metav1.NewTime(time.Now())
-			status.BasicAssessmentEndTime = &assessmentEndTime
+			status.TrialWindowStartTime = nil
+			status.AssessmentResult = apiv1.AssessmentResultFailure
 		})
 
 		return assessment, reasonFailure, nil
 	}
 
-	// if we succeed, we must continue to succeed for a prescribed period of time in order to consider the resource health
-	// check "successful"
+	// Handle success: set or keep TrialWindowStartTime
 	if assessment == apiv1.AssessmentResultSuccess {
-		// has AssessmentEndTime been set? if not, set it - now we can start our interval
-		if !childStatus.IsAssessmentEndTimeSet() {
+		if !childStatus.IsTrialWindowStartTimeSet() {
 			_ = progressive.UpdateUpgradingChildStatus(pipelineRollout, func(status *apiv1.UpgradingChildStatus) {
-				assessmentEndTime := metav1.NewTime(time.Now().Add(assessmentSchedule.Period))
-				status.BasicAssessmentEndTime = &assessmentEndTime
+				status.TrialWindowStartTime = &metav1.Time{Time: currentTime}
+				status.AssessmentResult = apiv1.AssessmentResultSuccess
```

Comment:
This should be AssessmentResultUnknown. Only once a full minute has occurred should it be AssessmentResultSuccess.